### ISSUE TITLE
Error while running ESLint: Unexpected token } in JSON at position 424.

### ISSUE THREAD

**Issue by createdbyjurand** [AtomLinter/linter-eslint#1045]
## Issue Description

I use Atom to verify js code.
I got first line on top underline red with this info.
I reinstalled all packages with yarn (before it was npm) - thought it could be some old dependiency.

.eslint config:
```json
{
  "plugins": [ "react" ],
  "extends": [ "eslint:recommended" ],
  "env": {
    "es6": true,
    "browser": true,
    "node": true,
    "jquery": true
  },
  "rules": { "strict": "off" },
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": { "jsx": true }
  }
}
```

## Bug Checklist
- Restart Atom
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Linter Eslint: Debug output here
```
Linter (bottom slide menu) gives the same error.

**Comment by skylize** [comment]
> Unexpected token } in JSON

That sounds like you have either an invalid `package.json` or invalid `.eslintconfig.json`. 

Make sure you have everything in double quotes (not single quotes), all your brackets matched, and no dangling commas. Remember that a json file should open and close with a curly brace. 

You can paste your json in the validator on [jsonlint.com](https://jsonlint.com/) for quick one-off linting.

**Comment by rj254** [comment]
I'm running into a similar issue except with `)` in the config. I'm using an eslintrc.js file though, and I have confirmed it is valid. The portion it errors out on is:

```
    rules: Object.assign(
        {},
        generalRules,
        flowRules,
        reactRules,
        prettierRules,
    )
```
where the error is Unexpected token `)`

**Comment by skylize** [comment]
@rj254 Your problem is only vaguely related. Please open a new Issue to track solving the problem on your system. I _very_ strongly suggest you include the output from `Linter Eslint: Debug` and your entire config file so we can actually see what's going wrong.

I _can_ tell you this much from your post: When an "Unexpected token" looks like it should be expected, that just means the bug is earlier in the code.

You have some code somewhere above there that the parser reads as valid until it runs into that closed paren and suddenly realizes what it thought you were doing doesn't make any sense.

**Comment by Arcanemagus** [comment]
@rj254 You have a trailing `,` after `prettierRules`. You may want to move to a `.eslintrc.js` file that exports your config.

**Comment by rj254** [comment]
Seems odd that the same configuration works both in the native `eslint` and in `vscode-eslint`. 

```
Atom version: 1.22.0
linter-eslint version: 8.4.0
ESLint version: 4.10.0
Hours since last Atom restart: 0
Platform: darwin
Using local project ESLint from:<PROJECT_ROOT>/node_modules/eslint
Current file's scopes: [
  "source.js.jsx"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by Arcanemagus** [comment]
Errors like that are forwarded on from ESLint itself, so I'm not sure why they aren't alerting you.

**Comment by rj254** [comment]
Running eslint doesn't yield any errors though. Are there restrictions on the `.eslintrc.js` that is specific to linter-eslint?

**Comment by Arcanemagus** [comment]
If while searching for a configuration file (to tell whether you have configured ESLint or not) and no `.eslint.___` file is found then this code is hit:
https://github.com/AtomLinter/linter-eslint/blob/7f791653480ce30c1ffe9ab99c6c84e34fe7e7fe/src/worker-helpers.js#L123-L125

So if there is a syntax error in your `package.json` then we could be throwing it from there. The only other way for that error to show up is to come from ESLint.

Without access to your files I can't really say more.

**Comment by IanVS** [comment]
@rj254, support for trailing commas in function argument lists [was added in node 8](http://node.green/#ES2017-features-trailing-commas-in-function-syntax-in-argument-lists).  I believe the version atom uses is significantly older than that.

**Comment by Arcanemagus** [comment]
@IanVS Atom (currently) bundles Node.js v7.4.0. Not _significantly_ older, but not new enough for that 😉.

**Comment by IanVS** [comment]
They're doing a better job keeping up than they used to, then!  Wasn't atom stuck at 0.12 until fairly recently?

**Comment by Arcanemagus** [comment]
Atom v1.9.0 had Electron v0.36.12 which had some version of Node.js <= 5.1.1. (Version 0.37.3 they added a `.node_version` file targeting v5.1.1, but that is only loosely coupled with the version of Node.js actually used.)

tl;dr: Atom hasn't used 0.12 for a _long_ time.

**Comment by IanVS** [comment]
Alright, my memory failed me I guess.  I thought it was like a year ago or so.  ¯\_(ツ)_/¯  Time flies.

**Comment by IanVS** [comment]
It seems that we've determined the cause of the errors as reported by the users in this issue, and I don't believe there are any changes needed in `linter-eslint`, so I'm closing this issue.  If anyone else feels differently, please speak up!

**Comment by nyngwang** [comment]
I also encounter this fucking comma :( waste some hours... Thanks for you post btw :)
